src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,89 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_CAPACITY = 10;
+    private int size = 0;
+    private T[] data = (T[]) new Object[ARRAY_CAPACITY];
 
     @Override
     public void add(T value) {
-
+        if (size == data.length) {
+            increaseCapacity();
+        }
+        data[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == data.length) {
```

Comment:
let’s also check the index - it should not be less than 0 or greater than size (it can be equal to size though)